projects/js-packages/publicize-components/src/hooks/use-page-has-focus/index.js

```diff
@@ -0,0 +1,32 @@
+import { useState, useEffect, useCallback } from '@wordpress/element';
+
+const hasFocus = () => document?.hasFocus() ?? false;
+
+/**
+ * Hook that returns whether the page has focus or not.
+ *
+ * @param {boolean} runOnRender - Whether to run the callback on render.
+ * @returns {boolean} Whether the page has focus or not.
+ */
+const usePageHasFocus = ( runOnRender = false ) => {
```

Comment:
I'm not sure this hook is working efficiently. A request happens on focus and then the connection test results  request is made, which results in a rerender and two more settings requests are made.

Could we use the library and debouncing that we do for the connection test results? [That uses the react-page-visibility library](https://github.com/Automattic/jetpack/blob/565c2117912e1fc41dc39401798d01f29d50e9b8/projects/js-packages/publicize-components/src/components/settings-button/index.js#L23) which is a component but I see it [does have a hook](https://www.npmjs.com/package/react-page-visibility)

Reply:
Sounds good, changed, now we are using that.